consensus/hotstuff/forks/forks.go

```diff
@@ -9,63 +9,71 @@ import (
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/module"
 	"github.com/onflow/flow-go/module/forest"
+	"github.com/onflow/flow-go/module/mempool"
 )
 
-// Forks implements HotStuff finalization logic
+// ErrPrunedAncestry is a sentinel error: cannot resolve ancestry of block due to pruning
+var ErrPrunedAncestry = errors.New("cannot resolve pruned ancestor")
+
+// ancestryChain encapsulates a block, its parent (oneChain) and its grand-parent (twoChain).
+// Given a chain structure like:
+//   b <~ b' <~ b*
+//
+// This data structure looks like:
+//   twoChain   oneChain     block
+//   [b<-qc_b]  [b'<-qc_b']  [b*]
```

Comment:
a bit confused what `b<-qc_b` means, is that `qc_b` qualifies block b?

Reply:
I meant "block b, and the QC qualifying block b", the same as https://github.com/onflow/flow-go/blob/1e0a30e7a9118aad96a706639e37ec955f562542/consensus/hotstuff/forks/blockQC.go#L8-L10